File: baekjoon/p03076/Main.java
package baekjoon.p03076;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<String> solve = solve(sc.nextInt(), sc.nextInt(), sc.nextInt(), sc.nextInt());

        for (String e : solve) {
            System.out.println(e);
        }
    }

    private static List<String> solve(int r, int c, int a, int b) {
        final String X = repeat("X", b);
        final String DOT = repeat(".", b);
        List<String> ret = new ArrayList<>();

        for (int i = 0; i < r; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j < c; j++) {
                if (i % 2 == 0 && j % 2 == 0 || i % 2 == 1 && j % 2 == 1) {
                    sb.append(X);
                } else {
                    sb.append(DOT);
                }
            }

            String line = sb.toString();
            for (int k = 0; k < a; k++) {
                ret.add(line);
            }
        }

        return ret;
    }

    private static String repeat(String str, int cnt) {
        StringBuilder sb = new StringBuilder();
        while (cnt-- > 0) {
            sb.append(str);
        }

        return sb.toString();
    }

}

File: baekjoon/p03076/Main.kt
package baekjoon.p03076

fun main() {
    val (r, c) = readLine()!!.split(" ").map { it.toInt() }
    val (a, b) = readLine()!!.split(" ").map { it.toInt() }
    println(solve(r, c, a, b).joinToString("\n"))
}

private fun solve(r: Int, c: Int, a: Int, b: Int): List<String> {
    val x = "X".repeat(b)
    val dot = ".".repeat(b)
    val ret = ArrayList<String>()

    for (i in 0 until r) {
        val sb = StringBuilder()
        for (j in 0 until c) {
            when (i % 2 == 0 && j % 2 == 0 || i % 2 == 1 && j % 2 == 1) {
                true -> sb.append(x)
                false -> sb.append(dot)
            }
        }

        val line = sb.toString()
        repeat(a) {
            ret.add(line)
        }
    }

    return ret
}